frontend/javascripts/dashboard/dataset/dataset_settings_data_tab.tsx

```diff
@@ -184,6 +184,8 @@ function SimpleDatasetForm({
             <Row gutter={48}>
               <Col span={24} xl={12}>
                 <FormItemWithInfo
+                  // The dataset name is not synced with the datasource.id.name in the advanced settings, because datasource.id represents a DataSourceId
+                  // where datasource.id.name represents the dataset's directoryName and not the dataset's name.
```

Comment:
@daniel-wer I added a comment to the dataset name form item explaining why this is not in sync with `datasource.id.name`